<!DOCTYPE HTML>
<!--
	Editorial by HTML5 UP
	html5up.net | @ajlkn
	Free for personal and commercial use under the CCA 3.0 license (html5up.net/license)
-->
<html>
	<head>
		<title>Softwares - Susmita Datta</title>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
		<link rel="stylesheet" href="assets/css/main.css" />
	</head>
	<body class="is-preload">

		<!-- Wrapper -->
			<div id="wrapper">

				<!-- Main -->
					<div id="main">
						<div class="inner">

							<!-- Header -->
								<header id="header">

								</header>

							<!-- Content -->
								<section>
									<header class="main">
										<h1>Softwares</h1>
									</header>
									
									<p><font size="4">Some R Packages or R Codes Generated Through Our Research Team</font></p>
									
									
									<!-- Start for one software -->
									<hr class="major" />

										<div class="row">
											<div class="col-4 col-5-large col-12-medium" align="center">
											<br>
												<span class="image object"><img src="images/noimage.jpg" alt="" width="100%"></span>
											</div>
											<div class="col-8 col-7-large col-12-medium">
												<p><h3><a href="https://github.com/zhangrunzhi886/CAMDA-2020" target="_blank">
												R code for CAMDA 2020 Metagenomic Geolocation Challenge
												</a></h3></p>
												<p><font size="3">
												The composition of microbial communities has been known to be location-specific. Investigating the microbial composition across different cities enables us to unravel city-specific microbial signatures and further predict the origin of unknown samples. As part of the CAMDA 2020 Metagenomic Geolocation Challenge, MetaSUB provided the whole genome shotgun (WGS) metagenomics data from samples across 28 cities along with non-microbial city data for 23 of these cities. We here implement feature selection, normalization, clustering and three methods of machine learning to classify cities based on their microbial compositions. Multilayer Perceptron obtained the best performance with an error rate of 19.60% based on the highest and the second-highest votes for the main dataset. Using the main dataset to predict the origins of samples from the mystery dataset, for the mystery samples from cities that belonged to the main dataset, error rates ranged from 18.18% to 72.7%; for the cities that did not belong to the main dataset, 57.7% of the test samples could be correctly labeled as “mystery” samples. Furthermore, we also predicted some of the non-microbial features for mystery samples from the cities that did not belong to main dataset to draw inferences and narrow the range of the possible sample origins using a multi-output multilayer perceptron.
												</font></p>
									
											</div>
										</div>
										
										<br>
										
									<!-- End for one software -->
									
									<!-- Start for one software -->
									<hr class="major" />

										<div class="row">
											<div class="col-4 col-5-large col-12-medium" align="center">
											<br>
												<span class="image object"><img src="images/noimage.jpg" alt="" width="100%"></span>
											</div>
											<div class="col-8 col-7-large col-12-medium">
												<p><h3><a href="https://github.com/zhangrunzhi886/CAMDA-2019" target="_blank">
												R code for CAMDA 2019 Metagenomic Forensic Challenge
												</a></h3></p>
												<p><font size="3">
												Composition of microbial communities can be location-specific, and the different abundance of taxon within location could help us to unravel city-specific signature and predict the sample origin locations accurately. In this study, the whole genome shotgun (WGS) metagenomics data from samples across 16 cities around the world and samples from another 8 cities were provided as the main and mystery datasets respectively as the part of the CAMDA 2019 MetaSUB “Forensic Challenge”. The feature selection, normalization, three methods of machine learning, PCoA (Principal Coordinates Analysis) and ANCOM (Analysis of composition of microbiomes) were conducted for both the main and mystery datasets.
											
												</font></p>
									
											</div>
										</div>
										
										<br>
										
									<!-- End for one software -->
									
									
									<!-- Start for one software -->
									<hr class="major" />

										<div class="row">
											<div class="col-4 col-5-large col-12-medium" align="center">
											<br>
												<span class="image object"><img src="images/scREhurdle.png" alt="" width="100%"></span>
											</div>
											<div class="col-8 col-7-large col-12-medium">
												<p><h3><a href="https://github.com/mnsekula/scREhurdle" target="_blank">
												scREhurdle: Identifying Differentially Expressed Genes with Single Cell RNA-Seq Data
												</a></h3></p>
												<p><font size="3">
												scREhurdle is an R package for detecting differentially expressed genes in discrete single-cell RNA sequencing data. This package interfaces with rstan and fits a mixed effect hurdle model on zero-inflated count data.
											
												</font></p>
												<li>Sekula, M., Gaskins, J., Datta, S. (2019) Detection of differentially expressed genes in discrete single cell RNA. Accepted in <i>Biometrics</i>, April 22, 2019. DOI: 10.1111/biom.13074</li>
											</div>
										</div>
										
										<br>
										
									<!-- End for one software -->
										
										
									<!-- Start for one software -->
									<hr class="major" />

										<div class="row">
											<div class="col-4 col-5-large col-12-medium" align="center">
											<br>
												<span class="image object"><img src="images/RankAggreg.png" alt="" width="100%"></span>
											</div>
											<div class="col-8 col-7-large col-12-medium">
												<p><h3><a href="https://cran.r-project.org/web/packages/RankAggreg/" target="_blank">
												RankAggreg: Weighted Rank Aggregation of Cluster Validation Measures
												</a></h3></p>
												<p><font size="3">
												RankAggreg performs aggregation of ordered lists based on the ranks using several different algorithms: Cross-Entropy Monte Carlo algorithm, Genetic algorithm, and a brute force algorithm (for small problems).
											
												</font></p>
												<li>Pihur, V.,Datta, S. and Datta, S. RankAggreg: an R package for weighted rank aggregation. <i>BMC Bioinformatics</i>, (2009).</li>
												<li>Pihur, V., Datta, S. and Datta, S. Weighted rank aggregation of cluster validation measures: A Monte Carlo cross-entropy approach. <i>Bioinformatics</i>, 23, 1607-1615 (2007).</li>
											</div>
										</div>
										
										<br>
									<!-- End for one software -->
										
										
									<!-- Start for one software -->
									<hr class="major" />

										<div class="row">
											<div class="col-4 col-5-large col-12-medium" align="center">
											<br>
												<span class="image object"><img src="images/clValid.png" alt="" width="100%"></span>
											</div>
											<div class="col-8 col-7-large col-12-medium">
												<p><h3><a href="https://cran.r-project.org/web/packages/clValid/index.html" target="_blank">
												clValid: Validation of Clustering Results
												</a></h3></p>
												<p><font size="3">
												clValid contains functions for validating the results of a clustering analysis. There are three main types of cluster validation measures available, “internal”, “stability”, and “biological”. The user can choose from nine clustering algorithms in existing R packages, including hierarchical, K-means, self-organizing maps (SOM), and model based clustering. In addition, we provide a function to perform the self-organizing tree algorithm (SOTA) method of clustering.

												</font></p>
												<li>Brock, G.,Pihur, V., Datta, S. and Datta, S. clValid: an R package for cluster validation. <i>Journal of Statistical Software</i>, 25, 4 (2008).</li>
												<li>Datta, S.  and Datta, S. Methods for evaluating clustering algorithms for gene expression data using a reference set of functional classes, <i>BMC Bioinformatics</i>, 7:397 (2006). <img src="images/highlyaccessed.png" alt=""></li>
												<li>Datta, S. and Datta, S. Comparisons and validation of statistical clustering techniques for microarray gene expression data. <i>Bioinformatics</i>, 19,  459-466 (2003).</li>
											</div>
										</div>
										<br>
									<!-- End for one software -->

                                </section>
						</div>
					</div>

				<!-- Sidebar -->
					<div id="sidebar">
						<div class="inner">


							<!-- Menu -->
								<nav id="menu">
									<header class="major">
										<h2>Menu</h2>
									</header>
									<ul>
										<li><a href="index.html">Homepage</a></li>
										<li><a href="research.html">Research Interests</a></li>
										<li><a href="publication.html">Publications</a></li>
										<li><a href="software.html">Softwares</a></li>
										<li><a href="students.html">Students</a></li>
										<li><a href="http://biostat.ufl.edu/", target="_blank">Department Website</a></li>
									</ul>
								</nav>

							<!-- Footer -->
								<footer id="footer">
									<p class="copyright">&copy; Susmita Datta. All rights reserved. <br>Design: <a href="https://html5up.net">HTML5 UP</a>.</p>
								</footer>

						</div>
					</div>

			</div>

		<!-- Scripts -->
			<script src="assets/js/jquery.min.js"></script>
			<script src="assets/js/browser.min.js"></script>
			<script src="assets/js/breakpoints.min.js"></script>
			<script src="assets/js/util.js"></script>
			<script src="assets/js/main.js"></script>

	</body>
</html>